vic/src/plugins/ext_mpi/ext_mpi_support.c

```diff
@@ -520,6 +520,17 @@ mpi_map_decomp_domain_basin(size_t   ncells,
             }
         }
     }
+    
+    free(node_ids);
+    free(basin_to_node);
+    
+    for(i = 0; i < basins.Nbasin; i++){
+        free(basins.catchment);
+    }
+    free(basins.Ncells);
+    free(basins.basin_map);
+    free(basins.catchment);
+    free(basins.sorted_basins);
 }
```

Comment:
Thoses free statements are causing segmentation faults in eg. rout_start -> nc_open

Reply:
Sorry, this was already changed in the function below (**mpi_map_decomp_domain_file**) but I forgot to do it here. Changed **free(basins.catchment)** to **free(basins.catchment[i])** in the iterator.